acceptance/Rakefile

```diff
@@ -308,66 +258,53 @@ EOS
-  task :extract_preserved_config do
-    generate_config_for_latest_hosts
-  end
-
   desc <<-EOS
 Run an acceptance test for a given node configuration and preserve the hosts.
 Defaults to an aio run, but you can set it to 'git' with TYPE='git'.
+This can also be accomplished for any beaker task with OPTIONS='--preserve-hosts'.
 #{USAGE}
   EOS
-  task :test_and_preserve_hosts => 'ci:check_env'  do
-    beaker_test(beaker_run_type, :preserve_hosts => 'always', :__preserve_config__ => true)
+  task :test_and_preserve_hosts => ['ci:check_env', 'ci:gen_hosts'] do
+    beaker_test(beaker_run_type, :preserve_hosts => 'always')
   end
 
-  desc "List acceptance runs from the past day which had hosts preserved."
-  task :list_preserved do
-    preserved = list_preserved_configurations
-    print_preserved(preserved)
+  desc "List preserved hosts from latest (default) or runs within given hours; e.g.: rake ci:list_hosts[24]"
+  task :list_hosts, :hours_ago do |t, args|
+    latest = 0
+    secs_ago = args[:hours_ago].to_f*60*60 || latest
+    print_hosts(preserved_hosts(secs_ago))
   end
 
   desc <<-EOS
-Shutdown and destroy any hosts that we have preserved for testing.  These should be reaped daily by scripts, but this will free up resources immediately.
-Specify a list of comma separated HOST_NAMES if you have a set of dynamic vcloud host names you want to purge outside of what can be grepped from the logs.
-You can go back through the last SECS_AGO logs.  Default is one day ago in secs.
+Release preserved hosts within [:hours_ago] (latest sut.log by default) back to vmpooler; e.g.: rake ci:kill_hosts[24]"
   EOS
-  task :release_hosts do
-    host_names = ENV['HOST_NAMES'].split(',') if ENV['HOST_NAMES']
-    secs_ago = ENV['SECS_AGO']
-    release_hosts(host_names, secs_ago)
-  end
-
-  task :destroy_preserved_hosts => 'ci:release_hosts' do
-    puts "Note: we are now releasing hosts back to the vcloud pooling api rather than destroying them directly.  The rake task for this is ci:release_hosts"
+  # FIXME: test with other hypervisors? this implies beaker should probably handle this stuff
+  task :kill_hosts, :hours_ago do |t, args|
+    latest = 0
+    secs_ago = args[:hours_ago].to_f*60*60 || latest
+    host_names = preserved_hosts(secs_ago)
+    release_hosts(host_names)
   end
 
   desc <<-EOS
 Rerun an acceptance test using the last captured preserved_config.yaml to skip provisioning.
-Or specify a CONFIG_NUMBER from `rake ci:list_preserved`.
-Defaults to an aio run, but you can set it to 'git' with TYPE='git'.
+One can set install-type using, e.g.: TYPE='git'.
   EOS
   task :test_against_preserved_hosts do
-    config_number = (ENV['CONFIG_NUMBER'] || 0).to_i
-    preserved = list_preserved_configurations
-    print_preserved(preserved)
-    config_path = preserved[config_number][0]
-
-    puts "Using ##{config_number}: #{config_path}"
+    if ENV['CONFIG_NUMBER']
+      puts "WARNING, config_number capability has bit-rotted. ignoring"
```

Comment:
maybe we could view this as a non-major boundary and this needs to be deprecated?